### PR TITLE
add function bitpositionToArray

### PR DESCRIPTION
**Pull request by MaxWk**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
add bitpositionToArray function

Detailed description / Documentation draft:
Function to get position of non zero bits
https://github.com/ClickHouse/ClickHouse/issues/23792

By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Functions/FunctionsCoding.h [hunk 1]
```diff
@@ -1504,6 +1504,92 @@ class FunctionBitmaskToArray : public IFunction
...
+
+            for (size_t row = 0; row < size; ++row)
+            {
+                UnsignedT x = vec_from[row];
+                while (x)
+                {
+                    UnsignedT y = x & (x - 1);
+                    UnsignedT bit = x ^ y;
+                    x = y;
+                    res_values.push_back(std::log2(bit));
```
Comment (alexey-milovidov) [inline src/Functions/FunctionsCoding.h:1558]: This is extremely slow.

Comment (MaxWk) [reply src/Functions/FunctionsCoding.h]: Hi alexey,I have update the way to get bitpodition，could you review it again

**Comment by kitaisreal** [comment]
@MaxWk please add documentation, fix style check, and add function test for this new function.

**Comment by MaxWk** [comment]
> @MaxWk please add documentation, fix style check, and add function test for this new function.

@kitaisreal could you help me check it again ,thanks

File: docs/en/sql-reference/functions/encoding-functions.md [hunk 2]
```diff
@@ -172,3 +172,6 @@ Accepts an integer. Returns a string containing the list of powers of two that t
 
 Accepts an integer. Returns an array of UInt64 numbers containing the list of powers of two that total the source number when summed. Numbers in the array are in ascending order.
 
+## bitpositionToArray(num) {#bitpositiontoarraynum}
+
+Accepts an integer. Returns an array of UInt64 numbers containing the list of positions of bit that equals 1. Numbers in the array are in ascending order.
```
Comment (qoega) [inline docs/en/sql-reference/functions/encoding-functions.md:177]: I guess we need note what differs if we pass unsigned and signed integer.

**Comment by abel-cheng** [comment]
Something more to do with this issue?

File: src/Functions/FunctionsCoding.h [hunk 3]
```diff
@@ -1504,6 +1504,95 @@ class FunctionBitmaskToArray : public IFunction
...
+            res_offsets.resize(size);
+            res_values.reserve(size * 2);
+
+            for (size_t row = 0; row < size; ++row)
+            {
+                UnsignedT x = vec_from[row];
+                int position = 0;
+                while (x)
+                {
+                    if (x & 1)
```
Comment (kitaisreal) [inline src/Functions/FunctionsCoding.h:1556]: @MaxWk cant we use bitscan forward here ?

Comment (MaxWk) [reply src/Functions/FunctionsCoding.h:1556]: Hi kitaisreal,I am a newbie in C++, I am not very familiar with this, can you help me finish this

Comment (kitaisreal) [reply src/Functions/FunctionsCoding.h:1556]: @MaxWk yes, sure.

**Review by kitaisreal — APPROVED** [review]

**Pull request merged**